<?xml version="1.0"?>
<launch>

    <node pkg="cob_robot_calibration_est" type="multi_step_cov_estimator.py" name="cal_cov_estimator_v0" output="screen" launch-prefix="time"
        args="/tmp/cal/cal_measurements.bag /tmp/cal" respawn="False"/>

    <group ns="calibration_config" clear_params="true">
        <!-- sensors -->
        <group ns="sensors">
            <param name="all" textfile="$(find cob_robot_calibration)/config/$(env ROBOT)/sensors.yaml" />
	    <param name="camera_parameter" value="$(find
		    cob_calibration_data)/$(env ROBOT)/calibration/cameras" />
 	 </group>

        <!-- initial system for first calibration step -->
        <param name="initial_system" textfile="$(find cob_robot_calibration)/config/$(env ROBOT)/system.yaml" />

        <!-- calibration steps -->
        <group ns="cal_steps">
	        <group ns="step0">
                <param name="free_params" textfile="$(find cob_robot_calibration)/config/$(env ROBOT)/free_0.yaml" />
                <param name="use_cov" type="bool" value="False" />
                <rosparam>
                  sensors:
                  - arm_chain
                  - kinect_rgb
                  - left
                  - right
                </rosparam>
                <param name="output_filename" type="string" value="result_step_1" />
            </group>
	        <group ns="step1">
                <param name="free_params" textfile="$(find cob_robot_calibration)/config/$(env ROBOT)/free_1.yaml" />
                <param name="use_cov" type="bool" value="False" />
                <rosparam>
                  sensors:
                  - arm_chain
                  - kinect_rgb
                  - left
                  - right
                </rosparam>
                <param name="output_filename" type="string" value="result_step_2" />
            </group>
 	        <group ns="step2">
                <param name="free_params" textfile="$(find cob_robot_calibration)/config/$(env ROBOT)/free_2.yaml" />
                <param name="use_cov" type="bool" value="False" />
                <rosparam>
                  sensors:
                  - arm_chain
                  - kinect_rgb
                  - left
                  - right
                </rosparam>
                <param name="output_filename" type="string" value="result_step_3" />
            </group>
        </group>
    </group>

</launch>
